src/components/Canvas.tsx

```diff
@@ -1,9 +1,32 @@
-const Canvas = () => {
+import { Element } from '../types/Common';
+
+interface Props {
+  elements: Element[];
+}
+
+const Canvas = ({ elements }: Props) => {
   return (
-    <main className="border border-red-700 h-full">
-      Canvas
-      <svg viewBox="0 0 120 120" xmlns="http://www.w3.org/2000/svg"></svg>
-    </main>
+    <svg
+      className="border-4 border-green-600"
+      preserveAspectRatio="xMinYMin meet" //for the SVG container to be on the entire screen, while the elements inside kept the proportions and x=0, y=0 viewBox started from the upper left corner
+      viewBox="0 0 1920 1080"
+      width="100%"
+      height="100%"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      {elements.map((block, i) => (
+        <block.type //flexible&dynemic rendering svg-elements
+          key={i}
+          x={block.x}
+          y={block.y}
+          width={block.width}
+          height={block.height}
+          stroke={block.stroke}
+          strokeWidth={block.strokeWidth}
+          fill={block.fill}
```

Comment:
you could do `{...block}` instead of listing every property separately

Reply:
🤝